Sparkle/SUAppcast.m

```diff
@@ -128,7 +128,7 @@ - (NSDictionary *)attributesOfNode:(NSXMLElement *)node
         if (!attrName) {
             continue;
         }
-        [dictionary setObject:[attribute stringValue] forKey:attrName];
+        [dictionary setValue:[attribute stringValue] forKey:attrName];
     }
```

Comment:
Hello, zorgiepoo.  The issue here is to clear clang warnings.  What matters is not whether -stringValue can _actually_ return nil in this case, but whether the clang static analyzer in Xcode 8 _thinks_ it can return nil :))

As I've always understood it, -setObject:forKey: does not accept nil as the first param, but setValue:forKey: does.  That is why changing -setObject: to -setValue: clears this particular warning.

Alternatively, we could assign stringValue to a local variable and test it for nil.  I just tested that, and clang is happy that way too.  What do you think?